lib/queue.js

```diff
@@ -90,20 +90,15 @@ const Queue = function Queue(name, url, opts) {
   }
 
   if (_.isString(url)) {
-    opts = _.extend(
-      {},
-      {
-        redis: redisOptsFromUrl(url)
-      },
-      opts
-    );
+    opts = {
+      ...opts,
+      redis: { ...redisOptsFromUrl(url), ...(opts || {}).redis }
+    };
   } else {
-    opts = url;
+    opts = url || {};
   }
 
-  opts = _.cloneDeep(opts || {});
```

Comment:
delete this code may cause some problem，becase `delete opts.redis.keyPrefix;` after this code.

Reply:
code example:
```
var Queue = require('bull');
var option = { redis: { host: '127.0.0.1', port: '6379', password: '', db: 1, keyPrefix: 'test' }};
var videoQueue = new Queue('video transcoding', {...option}); // prefix: test
var imageQueue = new Queue('image transcoding', {...option}); // prefix: bull
```

Reply:
yes, you are right, I overlooked it.